packages/ra-core/src/controller/list/useInfiniteListController.spec.tsx

```diff
@@ -45,6 +46,84 @@ describe('useInfiniteListController', () => {
         debounce: 200,
     };
 
+    const dataProvider = testDataProvider({
+        getList: jest.fn().mockImplementation((_resource, params) =>
+            Promise.resolve({
+                data: [{ id: 0 }, { id: 1 }].slice(
+                    0,
+                    params.pagination.perPage
+                ),
+                total: 2,
+            })
+        ),
+    });
+
+    describe('onSelectAll', () => {
+        const Children = params => (
+            <div>
+                <button onClick={() => params.onSelectAll()}>Select All</button>
+                <button onClick={() => params.onSelectAll({ limit: 1 })}>
+                    Limited Select All
+                </button>
+                <button onClick={() => params.onSelect([0])}>Select 0</button>
```

Comment:
Why don't you update the `Basic` story to contains the buttons you need for the tests? Would be cleaner than having a `children` prop and would allow us to manually reproduce the tests you added